DS/DSassignment1/ArrayList1.java

```
@@ -0,0 +1,203 @@
+	        
+	        for (int i = 0; i < data.length / 2; i++) {
+	            Object temp = data[i]; // swap numbers
+	            data[i] = data[data.length - 1 - i];
+	            data[data.length - 1 - i] = temp;
+	        }
+	        System.out.println("Array is reversed  " );
+	        
+	}
+	
+	 public static void add(int position, Object value) {
+			
+		 int flag=0;	
+		 size++;
+		if(position < size){
+			flag=1;
+		}
+		if(flag==1){
+		int index1=-1;
+			Object myStore1[]=new Object[data.length-position];
+			
+		if(size<defaultSize){
+			for(int j=position;j<size;j++){
+				index1++;
+				myStore1[index1]=data[j];	
+			}
+				
+			data[position]=value;
+			for(int k=0;k<index1;k++){
+				 position++;
+				data[position]=myStore1[k];
+			}
+			index++;
+			}
+		else{
+			for(int j=position;j<size;j++){
+				index1++;
+				myStore1[index1]=data[j];	
+			}
+			increaseCapacity();
+				
+			data[position]=value;
+			for(int k=0;k<index1;k++){
+				position++;
+				data[position]=myStore1[k];
+			}
+			index++;
+			}
+			}
+		else{
+			System.out.println("Array index out of bound");
+			size--;
+			}
+	 }
+	 
+	
+	 public static int retrieveFirstOccuerence(int position,Object value)
+	 {
+		 int index1 = 0,flag=-1;
+		  position++;
+		 for(int i=position; i < size; i++){
+		    	if(data[i]==value){
+		    		flag=0;
+		    		index1=i;
+		    		break;
+		    	}
+		  }
+		 if(flag==0){
+			return index1;
+		 }
+		 else{
+			return flag;
+		 }
+	 }
+
+	 public static int retrieveFirstIndexOfParticularItem(Object value) {
+			// TODO Auto-generated method stub
+			int index2 = 0,flag = -1;
+			for(int i = 0; i < size; i++){
+				if(data[i]==value){
+					flag=1;
+					index2=i;
+					break;
+				}
+			}
+			if(flag==1){
+				return index2;
+				
+			}
+			else{
+				return flag;
+			}
+		}
+	 
+	
+	 public static void remove(int position) {
+			// TODO Auto-generated method stub
+			if(position >= size){
+				System.out.println("Array index out of bound");
```

ACTION: You need to throw an exception rather than displaying an info